### ISSUE TITLE
'Unbound variable' error for class inheritance with global keyword

### ISSUE THREAD

**Issue by abelcheung** [microsoft/pylance-release#950]
## Environment data

-   Language Server version: 2021.2.2 (possibly irrelevant, all versions since 2021.1.1 known to have same issue)
-   OS and version: XXX
-   Python version (& distribution if applicable, e.g. Anaconda): XXX

## Description of issue

I use a somewhat... er... non-standard template for some files, intended to avoid any re-exporting of imported modules (see code sample below). It works fine so far until I try to include class inheritance into the game. Pylance would complain about all superclasses used this way, although the code itself runs fine.

Just wondering, if this is a false positive or a deliberate warning.

(As a side note, I'm aware that the standard way is to just mark everything as private with underscore prefix)

## Screenshot

![圖片](https://user-images.githubusercontent.com/83110/107880884-7718ba80-6ed9-11eb-8fe3-10614ffa6e91.png)

## Code sample

```python
# stub here
class A: ...
class B(A): ...
class C(B): ...

def _init():

    # most imports done here
    global A, B, C

    class A:
        def __init__(self) -> None:
            print('A')
    class B(A): pass
    class C(B): pass

_init()
del _init

c = C()
```

**Comment by erictraut** [comment]
Thanks for reporting the issue. This will be fixed in the next release of pylance.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.2.3, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202123-17-february-2021
